### PR TITLE
added conan2 recipe

### PR DESCRIPTION
**Pull request by MartyMcFlyInTheSky**
As I said, great library! I want to do some tests to see if we can use it in our production environment. Therefore I made some small additions / changes to make libfork work with conan2. I'll also push a recipe to conancenter if this PR is accepted. Feel free to ask questions.

### CODE REVIEW

**Comment by ConorWilliams** [comment]
Hi Marty, many thanks for your continued support and the PR!

I'm very keen to add conan support and getting libfork on conancenter would be fantastic.

I have been putting a lot of work into V3 of libfork which brings big performance improvements and many theoretical guarantees to the library. I've also completed an academic paper detailing how V3 works which I am currently working to get published.

Ideally I'd like to port this patch to V3 however, I am not planning on releasing V3 until the pre-print has been published so for now lets get this into the main branch.

File: CMakeLists.txt
Comment (ConorWilliams) [inline CMakeLists.txt:1]: Previously we set `CMAKE_INSTALL_INCLUDEDIR` only if this was the top-level build to allow packagers to override this setting, I am not super comfortable unconditionally hardcoding `CMAKE_INSTALL_INCLUDEDIR`, is this common practice?, can we wrap this in an `if(PROJECT_IS_TOP_LEVEL)`?

File: CMakeLists.txt [hunk 1]
```diff
@@ -40,14 +43,5 @@ if(NOT CMAKE_SKIP_INSTALL_RULES)
   include(cmake/install-rules.cmake)
 endif()
 
-# ---- Developer mode ----
```
Comment (ConorWilliams) [inline CMakeLists.txt:43]: Why are we unconditionally enabling dev mode?

File: test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.15)
+project(libfork_package_test CXX)
+
+# TODO: remove these lines
```
Comment (ConorWilliams) [inline test_package/CMakeLists.txt:4]: does this need removing?

Comment (MartyMcFlyInTheSky) [reply test_package/CMakeLists.txt:4]: Yes indeed

File: CMakeLists.txt (hunk 1 above)
Comment (MartyMcFlyInTheSky) [reply CMakeLists.txt:43]: I thought you'd check for BUILD_X variables again anyway, so on first glance it seemed unnecessary. However I had a second look now and I see that you also add some lint and spell targets, so I'll add it back.

**Comment by MartyMcFlyInTheSky** [comment]
> I have no experience with conan, what commands would I need to run to build the test package?

The test package is usually built as part of the validation stage of creating the package to make sure the package has been correctly created. To build it manually go to test_package subdirectory and in conanfile.py switch comments here:

```
        self.requires("libfork/2.1.1")
        #self.requires(self.tested_reference_str)
```
(The tested reference is usually injected during the package creation progress, so we need to do that manually in that case). Now you're ready to build using the workflow described in README.

Basically this should do it:

```
conan install . --build missing
conan build .
```
Make sure your conan profile defines a cpp20 ready compiler (I think it will be already set by default in your case).

**Comment by MartyMcFlyInTheSky** [comment]
> with hindsight maybe the `conan.py` needs dev-mode for the test targets?

Thx for merging. Atm you have two preconditions to enable test, benchmark & co: you have to set libfork_DEVELOPER_MODE and BUILD_TESTING, BUILD_BENCHMARKS, .. etc. We could just use one barrier for those and put only the targets behind libfork_DEVELOPER_MODE, whereas test and benchmarks etc.. could also be enabled by just setting the corresponding variable.
